tests/helpers/test_entityfilter.py

```diff
@@ -61,6 +107,49 @@ def test_with_include_domain_case4a():
     assert testfilter("sun.sun") is False
 
 
+def test_with_include_glob_case4a():
+    """Test case 4a - include and exclude specified, with included glob."""
+    incl_dom = {}
+    incl_glob = {"light.*", "sensor.*"}
+    incl_ent = {"binary_sensor.working"}
+    excl_dom = {}
+    excl_glob = {}
+    excl_ent = {"light.ignoreme", "sensor.notworking"}
+    testfilter = generate_filter(
+        incl_dom, incl_ent, excl_dom, excl_ent, incl_glob, excl_glob
+    )
+
+    assert testfilter("sensor.test")
+    assert testfilter("sensor.notworking") is False
+    assert testfilter("light.test")
+    assert testfilter("light.ignoreme") is False
+    assert testfilter("binary_sensor.working")
+    assert testfilter("binary_sensor.another") is False
+    assert testfilter("sun.sun") is False
+
+
+def test_with_include_domain_glob_filtering_case4a():
+    """Test case 4a - include and exclude specified, both have domains and globs."""
+    incl_dom = {"light"}
+    incl_glob = {"*working"}
+    incl_ent = {}
+    excl_dom = {"binary_sensor"}
+    excl_glob = {"*notworking"}
+    excl_ent = {"light.ignoreme"}
+    testfilter = generate_filter(
+        incl_dom, incl_ent, excl_dom, excl_ent, incl_glob, excl_glob
+    )
+
+    assert testfilter("sensor.working")
+    assert testfilter("sensor.notworking") is False
+    assert testfilter("light.test")
+    assert testfilter("light.notworking") is False
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5745133)